### PR TITLE
Use readthedocs for documentation

### LINKED ISSUE #475: Support for multiple version of the documentation
Maybe using [mike](https://github.com/jimporter/mike)?

Anyway we want to support from 4.0 onwards (we could think to have a 4.0alpha if we want to check how it works right now, with 2 versions)

### PR DESCRIPTION
**Pull request by qiaojunfeng**
Close #484
Close #485

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -66,3 +66,19 @@ jobs:
...
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+          cache: "pip"
+      - run: pip install -r docs/requirements.txt
+      - run: mkdocs build -s
+        working-directory: ./docs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ENABLE_MKDOCS_GIT_COMMITTERS: False
```
Comment (LecrisUT) [inline .github/workflows/main.yml]: What is the purpose of this job? It is better to have RTD build for PR, then both author and reviewer can see the changes. If it's to test for warnings, dead links, etc. it should be specified more explicitly.

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.12"
```
Comment (LecrisUT) [inline .readthedocs.yaml]:
```suggestion
    python: "3"
```

If you don't want to manually maintain the python version and use the latest version.

**Review by LecrisUT — APPROVED** [review]
LGTM :+1:

Another tip, in `actions/setup-python`, `python-version: 3.x` is equivalent with the latest released version.

(the project does not appear on readthedocs.org, but I guess the backend was not setup yet)

**Comment by qiaojunfeng** [comment]
> Another tip, in `actions/setup-python`, `python-version: 3.x` is equivalent with the latest released version.

Thanks, fixed as well.

> (the project does not appear on readthedocs.org, but I guess the backend was not setup yet)

I don't have enough permission to import the repo on RTD, already asked Giovanni to do this.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -66,3 +66,20 @@ jobs:
...
+    name: Validate mkdocs links
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          # use the latest stable version
+          python-version: "3.x"
+          cache: "pip"
+      - run: pip install -r docs/requirements.txt
```
Comment (LecrisUT) [inline .github/workflows/main.yml]: Oh great mkdocs is hit by the same issue that `sphinxcontrib-bibtex` has: https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/345

```suggestion
     # `pybtex` uses `pkg_resources` which is deprecated. Use workaround until upstream `mkdocs_bibtext`decides on a solution
     # https://bitbucket.org/pybtex-devs/pybtex/issues/169/replace-pkg_resources-with
      - run: pip install -r docs/requirements.txt setuptools
```

Comment (LecrisUT) [reply .github/workflows/main.yml]: Here's another issue you can follow: https://github.com/shyamd/mkdocs-bibtex/issues/228

Comment (qiaojunfeng) [reply .github/workflows/main.yml]: Indeed, now I remember why I fix the python version to `3.11` in the yaml.
And thanks for creating an issue in the `mkdocs-bibtex` repo!

**Review by giovannipizzi — APPROVED** [review]

**Pull request merged**